cmd/cloudx/project/get_test.go

```diff
@@ -22,20 +35,56 @@ func TestGetProject(t *testing.T) {
 
 func TestGetServiceConfig(t *testing.T) {
 	t.Run("service=kratos", func(t *testing.T) {
-		stdout, _, err := defaultCmd.Exec(nil, "get", "kratos-config", defaultProject, "--format", "json")
-		require.NoError(t, err)
-		assert.True(t, gjson.Get(stdout, "selfservice.flows.error.ui_url").Exists())
+		testhelpers.SetDefaultProject(t, defaultConfig, defaultProject)
+		t.Run("explicit project", func(t *testing.T) {
+			assert.Equal(t, defaultProject, testhelpers.GetDefaultProject(t, defaultConfig))
+
+			stdout, _, err := defaultCmd.Exec(nil, "get", "kratos-config", defaultProject, "--format", "json")
+			require.NoError(t, err)
+			assert.True(t, gjson.Get(stdout, "selfservice.flows.error.ui_url").Exists())
+		})
+		t.Run("default project", func(t *testing.T) {
+			assert.Equal(t, defaultProject, testhelpers.GetDefaultProject(t, defaultConfig))
+
+			stdout, _, err := defaultCmd.Exec(nil, "get", "kratos-config", "--format", "json")
+			require.NoError(t, err)
+			assert.True(t, gjson.Get(stdout, "selfservice.flows.error.ui_url").Exists())
+		})
```

Comment:
In all these places you now duplicated the testing logic, just for one additional flag. I think it's worth separating the general project selection logic from the test logic.

You could introduce a helper function like this:

```go
testhelpers.WithExplicitAndDefaultProject(defaultProject, func(t *testing.T, exec func(args...string) {
  stdout, _, err := exec("get", "kratos-config", "--format", "json")
  require.NoError(t, err)
￼ assert.True(t, gjson.Get(stdout, "selfservice.flows.error.ui_url").Exists())
)
```

`WithExplicitAndDefaultProject` would then call the function twice, once with an `exec` function that injects the `defaultProject` as the third parameter, and one that doesn't.

Maybe there's another way to abstract this, but either way we should not duplicate the testing logic too much ;).

Reply:
My personal preference is typically to value expressiveness and simplicity in tests higher than avoiding duplication, but I aggree, that there might be a little to much duplication now.
I'll have to think about how to best solve this issue.

Reply:
I think the solution can be straight forward (quickly hacked in github comment):

```go
func run(t *testing.T, useProject string)
    assert.Equal(t, defaultProject, testhelpers.GetDefaultProject(t, defaultConfig))

    agrs := []{"get", "oauth2-config", "--format", "json"}
    if useProject != "" {
      args = append(args, "--project", useProject)
    }
    stdout, _, err := defaultCmd.Exec(nil, agrs)
    require.NoError(t, err)
    assert.True(t, gjson.Get(stdout, "oauth2").Exists(), stdout)
}

t.Run("explicit project", run(t, "abcdef"))
t.Run("default project", run(t, ""))
```

Reply:
done